### DISCUSSION TITLE
How to pre-inject prompt (e.g. prompt engineering)?

### DISCUSSION THREAD

**Discussion by jamesbraza** [oobabooga/text-generation-webui#3934] [Q&A] [answered]
I am using [TheBloke/Llama-2-13B-chat-GGUF/llama-2-13b-chat.Q4_K_S.gguf](https://huggingface.co/TheBloke/Llama-2-13B-chat-GGUF/tree/main) with `server.py` like so:

```bash
python server.py --model llama-2-13b-chat.Q4_K_S.gguf --listen
```

The question is, how can one pre-inject context information to the LLM behind the scenes?

I want to inject the LLM with some context like "you are superman, and you are talking with lois lane" behind the scenes, before the conversation even begins with "Ask anything":

**Comment by oobabooga** [comment, marked as answer]
Put it in the "Context" field at Parameters > Instruction template in place of `Answer the questions`. Make sure to use Instruct mode for llama-2-chat models.
